app/routes.js

```diff
@@ -1,125 +1,84 @@
-var config = require('./config');
-var log = require('bole')('routes');
-var express = require('express');
-var join = require('path').join;
+const config = require('./config');
+const express = require('express');
+const join = require('path').join;
 
 
 exports.router = new express.Router();
 
-var routes = [].concat.apply([], config.apps.map(add_app_routes));
+const routes = merge(config.apps.map(app_name => {
 
-exports.url_for = function (view_name, args) {
+  return merge(require(join(__dirname, app_name, 'routes')).map(route => {
```

Comment:
Maybe it's because I'm not much familiar with this part of the code but I wonder if this `map` within a `map` can be a bit confusing.

Reply:
Even if not strictly necessary probably split in a couple of statements and use variables name to document.

Reply:
Yep, as discussed, it's a bit dense. I will try to simplify it.